### PR TITLE
Network cleanup 

### PR DESCRIPTION
**Pull request by nerzhul**
* Move Connection threads to dedicated files (divide cpp file by 2 & make it more readable)
* ConnectionReceiveThread::processPacket now uses function pointer table to route MT packet types
* Remove many packet buffers copy (performance improval)
* Various code style fixes

### CODE REVIEW

File: src/network/connectionthreads.cpp [hunk 1]
```diff
@@ -0,0 +1,1402 @@
...
+			continue;
+		}
+
+		float resend_timeout = dynamic_cast<UDPPeer *>(&peer)->getResendTimeout();
+		bool retry_count_exceeded = false;
+		for (Channel &channel : (dynamic_cast<UDPPeer *>(&peer))->channels) {
+			std::list<BufferedPacket> timed_outs;
+
+			if (dynamic_cast<UDPPeer *>(&peer)->getLegacyPeer())
+				channel.setWindowSize(g_settings->getU16("workaround_window_size"));
```
Comment (SmallJoker) [inline src/network/connectionthreads.cpp]: Undocumented setting.

File: src/network/connectionthreads.cpp [hunk 2]
```diff
@@ -0,0 +1,1402 @@
...
+				<< " has timed out."
+				<< " (source=peer->timeout_counter)"
+				<< std::endl;
+			// Add peer to the list
+			timeouted_peers.push_back(peer->id);
+			// Don't bother going through the buffers of this one
+			continue;
+		}
+
+		float resend_timeout = dynamic_cast<UDPPeer *>(&peer)->getResendTimeout();
```
Comment (SmallJoker) [inline src/network/connectionthreads.cpp]: `UDPPeer *udp_peer = dynamic_cast<UDPPeer *>(&peer);`, shortens and improves readability on 9 lines.

File: src/network/connectionthreads.cpp [hunk 3]
```diff
@@ -0,0 +1,1402 @@
...
+
+		/* send acks immediately */
+		if (packet.ack) {
+			rawSendAsPacket(packet.peer_id, packet.channelnum,
+				packet.data, packet.reliable);
+			peer->m_increment_packets_remaining =
+				MYMIN(0, peer->m_increment_packets_remaining--);
+		} else if (
+			(peer->m_increment_packets_remaining > 0) ||
+				(stopRequested())) {
```
Comment (SmallJoker) [inline src/network/connectionthreads.cpp]: Indents missing and redundant parentheses.

File: src/network/connectionthreads.cpp (hunk 1 above)
Comment (nerzhul) [reply src/network/connectionthreads.cpp]: yes, i moved the code but didn't see that, what is that thing, i think i will remove it, it's too dangerous for a end user

File: src/network/connectionthreads.cpp (hunk 2 above)
Comment (nerzhul) [reply src/network/connectionthreads.cpp]: exact :)

**Pull request merged**